src/main/kotlin/io/omnitrade/client/OmnitradeClient.kt

```diff
@@ -0,0 +1,407 @@
+    }
+
+    @JvmName("setAccessKey")
+    fun accessKey(accessKey: String): Builder {
+      Builder.accessKey = accessKey
+      return this
+    }
+
+    @JvmName("setSecretKey")
+    fun secretKey(secretKey: String): Builder {
+      Builder.secretKey = secretKey
+      return this
+    }
+
+    @JvmName("setEndpoint")
+    fun endpoint(endpoint: String): Builder {
+      Builder.endpoint = endpoint
+      return this
+    }
+
+    @JvmName("setTimeout")
+    fun timeout(timeout: Long): Builder {
+      Builder.timeout = timeout
+      return this
+    }
+  }
+
+  private val auth: Auth by dependencies.instance(arg = M(accessKey, secretKey))
+  private val omnitradeRepository: OmnitradeRepository by dependencies.instance(arg = M(endpoint, timeout))
+
+  // Public API Calls
+
+  fun getMarkets(): CompletableFuture<List<Market>> {
+    return omnitradeRepository.getMarkets()
+  }
+
+  fun getOrderBook(market: Market, asksLimit: Int, bidsLimit: Int): CompletableFuture<OrderBook> {
+    val options = mutableMapOf(
+      "market" to market.id,
+      "asks_limit" to asksLimit.toString(),
+      "bids_limit" to bidsLimit.toString()
+    )
+
+    return omnitradeRepository.getOrderBook(options).withExceptionOnError()
+  }
+
+  fun getDepth(market: Market, limit: Int): CompletableFuture<Depth> {
+    val options = mutableMapOf(
+      "market" to market.id,
+      "limit" to limit.toString()
+    )
+
+    return omnitradeRepository.getDepth(options).withExceptionOnError()
+  }
+
+  fun getOrderMarket(market: Market, marketType: MarketType, volume: Double): CompletableFuture<OrderMarket> {
+    val options = mutableMapOf(
+      "market" to market.id,
+      "type" to marketType.type,
+      "volume" to volume.toString()
+    )
+
+    return omnitradeRepository.getOrderMarket(options).withExceptionOnError()
+  }
+
+  fun getTrades(market: Market, tradeIdSince: Int? = null, tradeIdUntil: Int? = null, timestamp: Long? = null, limit: Int? = null, orderBy: OrderBy): CompletableFuture<List<Trade>> {
+    val options = mutableMapOf(
+      "market" to market.id,
+      "order_by" to orderBy.order
+    )
+
+    tradeIdSince?.let { options["since"] = it.toString() }
+    tradeIdUntil?.let { options["until"] = it.toString() }
+    timestamp?.let { options["timestamp"] = it.toString() }
+    limit?.let { options["limit"] = it.toString() }
+
+    return omnitradeRepository.getTrades(options).withExceptionOnError()
+  }
+
+  fun getK(market: Market, limit: Int, period: Period, timestamp: Long? = null): CompletableFuture<List<List<Int>>> {
+    val options = mutableMapOf(
+      "market" to market.id,
+      "limit" to limit.toString(),
+      "period" to period.time.toString()
+    )
+
+    timestamp?.let { options["timestamp"] = it.toString() }
+
+    return omnitradeRepository.getK(options).withExceptionOnError()
+  }
+
+  fun getKWithPendingTrade(market: Market, trade: Trade, limit: Int, period: Period): CompletableFuture<KWithPendingTrade> {
```

Comment:
Should the name of this method be in the plural as well?

Reply:
It should! Great catch!